Android/PushTutorial/app/src/main/java/io/ably/example/androidpushexample/MainActivity.java

```diff
@@ -1,61 +1,174 @@
+                return;
+            }
+
+            logMessage("Device is now registered for push");
+            handler.sendMessage(handler.obtainMessage(SUCCESS, STEP_3));
+        }
+    };
+
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
-        initAblyRuntime();
+        setContentView(R.layout.activity_main);
+        rollingLogs = findViewById(R.id.rolling_logs);
+        stepsButton = findViewById(R.id.steps);
     }
 
-    private void initAblyRuntime() {
-        try {
-            ClientOptions options = new ClientOptions();
-            options.environment = BuildConfig.ABLY_ENV;
-            options.key = BuildConfig.ABLY_KEY;
-
-            ablyRealtime = new AblyRealtime(options);
-            ablyRealtime.setAndroidContext(getApplicationContext());
-            ablyRealtime.connect();
-
-            ablyRealtime.connection.on(new ConnectionStateListener() {
-                @Override
-                public void onConnectionStateChanged(ConnectionStateChange state) {
-                    logMessage("Connection state changed to : " + state.current.name());
-                    switch (state.current) {
-
-                        case connected:
-                            break;
-                        case disconnected:
-                        case failed:
-                            //do something
-                            break;
-
-                    }
+
+    private void initAblyRuntime() throws AblyException {
+
+        ClientOptions options = new ClientOptions();
+        options.environment = BuildConfig.ABLY_ENV;
+        options.key = BuildConfig.ABLY_KEY;
+
+        ablyRealtime = new AblyRealtime(options);
+        ablyRealtime.setAndroidContext(getApplicationContext());
+        ablyRealtime.connect();
+
+        ablyRealtime.connection.on(new ConnectionStateListener() {
+            @Override
+            public void onConnectionStateChanged(ConnectionStateChange state) {
+                logMessage("Connection state changed to : " + state.current.name());
+                switch (state.current) {
+                    case connected:
+                        //Go to step 2
+                        handler.sendMessage(handler.obtainMessage(SUCCESS, STEP_2));
+                        break;
+                    case disconnected:
+                    case failed:
+                        handler.sendMessage(handler.obtainMessage(FAILURE));
+                        break;
+
                 }
-            });
-        } catch (AblyException e) {
-            e.printStackTrace();
-        }
+            }
+        });
+
+    }
+
+    private void initAblyPush() throws AblyException {
+        LocalBroadcastManager.getInstance(this).registerReceiver(pushReceiver, new IntentFilter("io.ably.broadcast.PUSH_ACTIVATE"));
+        ablyRealtime.push.activate();
+    }
+
+    private void subscribleChannels() {
```

Comment:
The typo here is corrected in a later commit, but please can we fix it in this commit?